### ISSUE TITLE
Add support for magrittr

### ISSUE THREAD

**Issue by g4challenge** [yihui/formatR#54]
Please add reformatting for magrittr pipe operator `%>%`.

For example:
`
result <- mydata %>% filter(criterion) %>% group_by(mygroup) %>% summarise(summand = sum(count))`

should be converted to something like:

```
mydata %>%
  filter(criterion) %>%
    group_by(mygroup) %>%
      summarise(summand = sum(count)) -> results
```

**Comment by edlee123** [comment]
Another example:
```
testfunc = function(){
  
  testout <- mtcars %>%
    mutate(mpgXcyl = mpg * cyl) %>%
    arrange(desc(disp), hp) %$% 
    carb
  return(testout)
}
```

```
> tidy_source()
Error in parse(text = lines, keep.source = TRUE) : 
  <text>:8:0: unexpected end of input
6:     carb
7:   return(testout)
  ^
> 
```

**Comment by yihui** [comment]
My reply would be the same as in #53...

**Comment by yihui** [comment]
@g4challenge This has been implemented. You can install the development version via

```r
remotes::install_github('yihui/formatR')
```

Thanks!

@edlee123 This turns out to be a different bug, which I fixed in ff21861.
> `> tidy_source()
> Error in parse(text = lines, keep.source = TRUE) : 
>   <text>:8:0: unexpected end of input
> 6:     carb
> 7:   return(testout)
>   ^`
